libs/libvtrutil/src/vtr_memory.h

```diff
@@ -136,6 +136,11 @@ struct aligned_allocator {
     }
 };
 
+template<typename T>
```

Comment:
mmh, so an comparison operator that always asserts that both its parameters are equal sounds like calling for trouble. Why is it impossible to compare these ?

If this is indeed intented, then it requires a good explanation why it is safe in this case.

Reply:
The allocator doesn't have any state, so there is no way for any one "instance" to be different from another one. The equivalent in `std::allocator` is even more generic and also describes allocators for different types as equal, which may or may not be appropriate here as well: https://github.com/gcc-mirror/gcc/blob/releases/gcc-11.1.0/libstdc%2B%2B-v3/include/bits/allocator.h#L206

Reply:
So add the information to the comment that there is no state to compare.

Reply:
Done.